src/colleges.py

```diff
@@ -22,8 +22,12 @@
     source="https://beachpartyserver.azurewebsites.net/VueBigData/DataFiles"
 )
 
-data_dir = fetch(COLLEGES_CONFIG)
-file = os.listdir(data_dir)[0]
-csv_path = os.path.join(data_dir, file)
 
-df = pandas.read_csv(csv_path, sep='\t', encoding='latin1')
+def get_colleges_df(save=True):
+    data_dir = fetch(COLLEGES_CONFIG)
+    file = os.listdir(data_dir)[0]
+    csv_path = os.path.join(data_dir, file)
+    df = pandas.read_csv(csv_path, sep='\t', encoding='latin1')
```

Comment:
It seems that `'UNITID'` would be the index, isn't it:

```python
df = pandas.read_csv(csv_path, sep='\t', encoding='latin1', index='UNITID')
```

Reply:
You also need to drop the `'Unnamed: 0'` column using `df.drop`

Reply:
In addition, we can tag the categorical column:

df['State'] = df['State'].astype(str).astype('category')

When doing so, all values in this column shoul be encoded as a string first. It is required by the liac-arff.